pkg/zpdb/eviction_controller.go

```diff
@@ -23,6 +28,12 @@ const (
 	logAllowMesg = "pod eviction allowed"
 )
 
+type IEvictionController interface {
+	MarkPodAsDeleted(ctx context.Context, namespace string, podName string, source string) error
+}
+
+var _ IEvictionController = (*EvictionController)(nil)
```

Comment:
The convention for interfaces like this that are used as a way to break a dependency on a concrete type is to define them in the consuming package, unless they're widely used.

So I'd move this interface to `pkg/controller` and call it something like `ZPDBEvictionController`.